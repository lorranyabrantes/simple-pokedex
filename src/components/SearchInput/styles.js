import { StyleSheet } from "react-native";
import { COLORS, BORDER_RADIUS, SPACING } from "../../services/styles";

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    backgroundColor: COLORS.GRAY,
    borderRadius: BORDER_RADIUS.medium,
    paddingHorizontal: SPACING.medium,
  },
  input: {
    height: 40,
    flexGrow: 1,
  },
  icon: {
    width: 20,
    height: 20,
  },
});

export default styles;
