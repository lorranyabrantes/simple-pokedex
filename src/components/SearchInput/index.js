import React, { useState, useCallback, useEffect } from "react";
import { View, TextInput, Image } from "react-native";

import iconSeach from "../../assets/png/icon-search.png";

import styles from "./styles";

const SearchInput = ({ onChangeText, value }) => {
  const [text, setText] = useState(value);

  const changeText = useCallback(
    (newText) => {
      setText(newText);
      onChangeText(newText);
    },
    [text, onChangeText]
  );

  useEffect(() => {
    setText(value);
  }, [value]);

  return (
    <View style={styles.container}>
      <TextInput
        testID={"seach-input"}
        style={styles.input}
        placeholder="Buscar"
        onChangeText={(value) => changeText(value)}
        value={text}
      />

      <Image style={styles.icon} source={iconSeach} />
    </View>
  );
};

export default SearchInput;
