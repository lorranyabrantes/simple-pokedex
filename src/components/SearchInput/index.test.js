import React from "react";
import { render } from "@testing-library/react-native";
import SearchInput from "./";

const props = {
  onChangeText: jest.fn(),
  value: "",
};

describe("SearchInput", () => {
  test("should render correctly", async () => {
    const component = render(<SearchInput {...props} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
