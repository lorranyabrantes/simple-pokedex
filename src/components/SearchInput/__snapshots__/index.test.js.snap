// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#CCCCCC",
      "borderRadius": 5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 16,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Buscar"
    rejectResponderTermination={true}
    style={
      Object {
        "flexGrow": 1,
        "height": 40,
      }
    }
    testID="seach-input"
    underlineColorAndroid="transparent"
    value=""
  />
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/png/icon-search.png",
      }
    }
    style={
      Object {
        "height": 20,
        "width": 20,
      }
    }
  />
</View>
`;
